interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX is the coding format for React that allows the code to be written in html and javascript. a syntax difference is the use of className to modify an element in JSX unlike html whihc uses class to modify an element. 
 
-Researched answer:
+Researched answer: JSX is a syntax extension of Javascript which behaves very similarly to html, It also uses a syntax called classNAme instead of class like html to modift elements because class is a protected javascript syntax word.
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is a package manager for jest which allows the user to run tests on code to ensure their functionality. when Yarn is ren in ther terminal it will modify the node module, package .Json, yarn.lock and a readme file.
 
-Researched answer:
+Researched answer: Yarn is a package manager that allows Jest to be ran for testing, it also replaces other package managers and was created as a collaboration of tech companies such as Google, Facebook and Expo . dec. (It began being used in 2016)
 
 3. What is a React component?
 
-Your answer:
+Your answer: Code that relays to a browser and displays an output.
 
-Researched answer:
+Researched answer: All code within React can be considered a React Component. these components are made of Javascript logic and are able to return markup that transfers to a browser creating a functional page.There is more than one way to create react components.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: The difference is that Props are static and cannot be changed but state values can be updated to hold a new value.
 
-Researched answer:
+Researched answer: A prop is an object that passes along information between a component and a nested component and are static once set not allowing them to be changed, however a State value is a dynamic variable and can be updated to hold a new value. 
 
 5. What is the DOM?
 
-Your answer:
+Your answer: a tree structured representation of a web pages html.
 
-Researched answer:
+Researched answer: A structure model that shows the Full html layout of a webpage, it represents a users interface.
 
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer: A div tag can be described as a container that can hold some code to keep it seperate from other html code allowing it to be selected and styled in CSS. a span tag allows you to select specific code to then be selected and styled in CSS.
 
-Researched answer:
+Researched answer: A div tag is a container that groups sets of content together on a webpage used in HTML. A span tag is used to contain a part of a document or part of mark uo text (an Inline Conatiner)
```

Comment:
A span tag will allow you to get even more specific with what you want to style. For instance, a couple of words within a p tag.